tests/test_mongodb_incremental.py

```diff
@@ -318,16 +394,56 @@ def test_run(self):
                                                                    self.expected_sync_streams(),
                                                                    self.expected_pks())
 
-        # Verify that we got 3 records for each stream (2 because of the new records, 1 because
-        # of gte)
-        for k,v in record_count_by_stream.items():
-            self.assertEqual(3, v)
+        # Verify that we got 4 records for each stream (2 because of the new records, 1 because of update and 1 because of greater than equal [for key based incremental there will always be an overlap on the bookmark value])
+        for k, v in record_count_by_stream.items():
+            # Workaround for not including collections for uuid and string, TODO : look for a solution to implement string and uuid as replication_key
+            if k not in ('simple_coll_uuid_field', 'simple_coll_string_field'):
+                self.assertEqual(4, v)
 
         # Verify that the _id of the records sent are the same set as the
         # _ids of the documents changed
         for stream_name in self.expected_sync_streams():
-            actual = set([x['data']['int_field'] for x in records_by_stream[stream_name]])
-            self.assertEqual(self.expected_incremental_int_fields()[stream_name], actual)
+            # Workaround for not including collections for uuid and string, TODO : look for a solution to implement string and uuid as replication_key
```

Comment:
Could you card this out as future work with label `tap-tester-upgrades1

Reply:
https://jira.talendforge.org/browse/TDL-16915